### PR TITLE
Add December 2024 newsletter to the website

### LINKED ISSUE #1301: Add December 2024 newsletter to the website
## Description

Every month, we try to get the newsletter up on the site within a week of emailing it. Currently, we're moving them over "by hand."

## Steps to Update

In the code base, navigate to `content > newsletters` and create a new file `2024-12.jsx`.
You can look at the existing newsletters as a template.

Make sure to add it to the index by following the steps in the ["Newsletters" section in our README](https://github.com/Virtual-Coffee/virtualcoffee.io#newsletters) and update the content accordingly based on our email newsletter.

### PR DESCRIPTION
**Pull request by adiati98**
## Linked Issue

Closes #1301 

## Description

This PR adds December 2024 newsletter to the website and updates the data in the `newsletters.ts` file.

## Code of Conduct

> By submitting this pull request, you agree to follow our [Code of Conduct](https://virtualcoffee.io/code-of-conduct/)

### CODE REVIEW

**Comment by adiati98** [comment]
@danieltott @BekahHW @meg-gutshall , this PR is ready for review. Thanks! 😊

File: src/content/newsletters/2024-12.jsx [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+			</p>
+			<p>
+				Devs are more than just the code we write. This challenge is all about
+				embracing self-expression. Give back to yourself by indulging in
+				something just for fun. Share the art, music, poetry, sports, games, or
+				other hobbies that spark your joy. We spend so much time grinding away
+				on understanding things in the tech space. Let's make some space for the
+				other parts of ourselves. In this challenge, we encourage you to spend
+				time working on things that aren't necessarily code-specific or using
+				code to improve your other hobbies and outlets.
```
Comment (meg-gutshall) [inline src/content/newsletters/2024-12.jsx:91]: **suggestion:** Let's throw some exclamation points in there!

```suggestion
				Devs are more than just the code we write. This challenge is all about
				embracing self-expression. Give back to yourself by indulging in
				something just for fun! Share the art, music, poetry, sports, games, or
				other hobbies that spark your joy. We spend so much time grinding away
				on understanding things in the tech space. Let's make some space for the
				other parts of ourselves! In this challenge, we encourage you to spend
				time working on things that aren't necessarily code-specific or using
				code to improve your other hobbies and outlets.
```

File: src/content/newsletters/2024-12.jsx [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+					Wednesdays at 10:00 PM ET)
+				</li>
+				<li>
+					Data Structures and Algorithms (DSA) Office Hours (Wednesdays at 4:00
+					PM ET)
+				</li>
+				<li>Feelings Friday (Fridays at 8:00 PM ET)</li>
+				<li>
+					Frontend Friday Folks fighting CSSBattle.dev (Fridays at 11:00 AM ET)
+				</li>
```
Comment (meg-gutshall) [inline src/content/newsletters/2024-12.jsx:125]: **nitpick:** Can we swap the order of these two?

```suggestion
				<li>
					Frontend Friday Folks fighting CSSBattle.dev (Fridays at 11:00 AM ET)
				</li>
				<li>Feelings Friday (Fridays at 8:00 PM ET)</li>
```

File: src/content/newsletters/2024-12.jsx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+					chat and gratitude post in <code>#general</code>
+				</li>
+			</ul>
+			<ul>
+				<li>
+					<a href="https://github.com/orgs/Virtual-Coffee/discussions/1223">
+						Practical Deep Learning Study Group
+					</a>
+				</li>
+			</ul>
```
Comment (meg-gutshall) [inline src/content/newsletters/2024-12.jsx:152]: **issue:** This is no longer happening. Please remove.

```suggestion
```

File: src/content/newsletters/2024-12.jsx [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+				events at the time of this publication. Please check the official VC{' '}
+				<code>#announcements</code> Slack channel, or other noted channels, for
+				any updates and links to event rooms. For the full list of events, check
+				out <Link href="/events">our events page</Link>.
+			</p>
+
+			<h3 className="mt-5 mb-4">🎙️ The Podcast is Back!</h3>
+			<p>
+				Check out our latest podcasts from season 10{' '}
+				<a href="https://virtualcoffee.io/podcast">here</a>!
```
Comment (meg-gutshall) [inline src/content/newsletters/2024-12.jsx:175]: **question:** Is there a better way to represent this link? I've been taught that linking words like "here" is a bad practice for accessibility reasons.

**Review by meg-gutshall — CHANGES_REQUESTED** [review]
Thanks, Ayu! Please see my comments. I know you just copy and paste from the newsletter, but I suggested some adjustments.

File: src/content/newsletters/2024-12.jsx (hunk 4 above)
Comment (adiati98) [reply src/content/newsletters/2024-12.jsx:175]: Nice catch! I'll fix this!

**Review by meg-gutshall — APPROVED** [review]
Thanks Ayu! Looks great.

**Pull request merged**